File: app/src/main/java/helper/StringToArray.java
package helper;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

public class StringToArray {

    public static List<String> convertStringToArrayCommaSeparated(String notSeparatedString) {
        String[] commaSeparatedArr = notSeparatedString.split("\\s*,\\s*");
        return new ArrayList<String>(Arrays.asList(commaSeparatedArr));
    }

    public static List<String> convertStringToArraySpaceSeparated(String commaSeparatedString) {
        String[] spaceSeparatedArr = commaSeparatedString.split("\\s+");
        return new ArrayList<String>(Arrays.asList(spaceSeparatedArr));
    }
}

File: app/src/main/java/helper/StringToArray.kt
package helper

import java.util.ArrayList
import java.util.Arrays

object StringToArray {

    fun convertStringToArrayCommaSeparated(notSeparatedString: String): List<String> {
        val commaSeparatedArr = notSeparatedString.split("\\s*,\\s*".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()
        return ArrayList(Arrays.asList(*commaSeparatedArr))
    }

    fun convertStringToArraySpaceSeparated(commaSeparatedString: String): List<String> {
        val spaceSeparatedArr = commaSeparatedString.split("\\s+".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()
        return ArrayList(Arrays.asList(*spaceSeparatedArr))
    }
}
